samples/23.react-live-canvas/src/useLiveCanvas.jsx

```diff
@@ -0,0 +1,43 @@
+import { InkingManager } from "@microsoft/live-share-canvas";
+import { useCallback, useEffect, useState } from "react";
+
+export const useLiveCanvas = (fluidContainer, hostingElement) => {
+  const [inkingManager, setInkingManager] = useState(undefined);
+
+  const handleBlackPenClick = useCallback(() => {
+    inkingManager.penBrush.color = { r: 0, g: 0, b: 0 }
+  }, [inkingManager])
+
+  const handleBluePenClick = useCallback(() => {
+    inkingManager.penBrush.color = { r: 0, g: 0, b: 255, a: 1 }
+  }, [inkingManager])
+  
+  const handleClearClick = useCallback(() => {
+    inkingManager.clear();
+  }, [inkingManager])
+
+  const startInkingManager = useCallback(async () => {
+    if (!fluidContainer || !hostingElement) {
+        return;
+    }
+    const inkingHost = hostingElement;
+    const inkManager = new InkingManager(inkingHost);
+    const liveCanvas = fluidContainer.initialObjects.liveCanvas;
+    await liveCanvas.initialize(inkManager);
+
+    // Activate the InkingManager so it starts handling pointer input
+    inkManager.activate();
+    setInkingManager(inkManager);
+  }, [fluidContainer, hostingElement]);
+
+  useEffect(() => {
+    startInkingManager();
+  }, [startInkingManager]);
+
+  return {

```

Comment:
It seems to me that the demo would be all the more useful if `useLiveCanvas` returned the `InkingManager` instance. Otherwise, `useLiveCanvas` has to act as some sort of a proxy to all the features/APIs exposed by `InkingManager`.

Reply:
Yes, it does return inking manager along with the helper callbacks.

Reply:
What I meant is that you don't need to expose `setToHighlighter`, `clearCanvas` etc. since these APIs are provided by the `InkingManager` instance.

Reply:
as per discussion, moved to the component.